### ISSUE TITLE
Takepos / Multi-compagny - Third-party replication problem

### ISSUE THREAD

**Issue by paverneuil** [Dolibarr/dolibarr#31537]
### Bug

Hello,
We have just detected a takepos and multi compagny incompatibility. When a third party is identified in the takepos interface, it is replicated on all other entities.

We've deactivated all caches, and performed tests on two different servers.

**entities A :**
![image](https://github.com/user-attachments/assets/1cb47c45-a985-4a1a-acf3-0d9aee275838)

**entities B :**
![image](https://github.com/user-attachments/assets/a80db35c-35f5-4875-8ddc-c3a1d5401a4f)

### Dolibarr Version

18.0.5

### Environment PHP

8.1

### Environment Database

8.0

### Steps to reproduce the behavior and expected behavior

1. Create two different entities with takepos enabled and configured on each.
2. Open the takepos entities page on two different workstations.
3. In entity A, identify a customer in takepos
4. In entity B, press F5 to refresh the page, you will see that the user from entity A will be identified on page B.

**Comment by hregis** [comment]
@paverneuil yes you right, the "terminal 1" as the same name in llx_const for all entities : "TAKEPOS_TERMINAL_NAME_1" and takepos check the number "1" for terminal but not the entity... For the moment try to create a second terminal in entity 2 ans use this "terminal 2" in entity 2

**Comment by hregis** [comment]
@paverneuil This problem is complicated to solve because many parameters are stored and read as session variables throughout the TakePOS code. To counter this, I've integrated into the latest versions of Multicompany the ability to associate a subdomain with each entity so that a new browser tab and a different session open when switching to another entity.

**Comment by paverneuil** [comment]
Bonjour,
Pourriez-vous m'indiquer si cette option est disponible en dernière version compatible dolibarr v18 ? Ou devons nous prévoir un upgrade ? nous avons essayé de tester l'option sur votre instance de test, mais nous n'avons pas trouvé l'option.

**Comment by hregis** [comment]
@paverneuil 
cette option est présente à partir de Multicompany 19.0.3 et il faut un Dolibarr 19
c'est encore une option cachée qui nécessite de mettre la valeur 1 à la constante `MULTICOMPANY_FEATURES_LEVEL` dans "Configuration -> divers"

**Comment by hregis** [comment]
@paverneuil 

une fois l'option activée un champ "url" est disponible dans la configuration des entités dans le module.
par contre ça nécessite aussi de créer des enregistrement A dans le DNS du nom de domaine.

<img width="923" height="540" alt="Image" src="https://github.com/user-attachments/assets/0e0d388c-ce46-4757-9573-282be8374f1e" />
